harrisonDir/harrisonHomework.txt

```diff
@@ -0,0 +1,113 @@
+PART 1:
+
+a) true, because == treats strings and numbers as being equal
+
+b) false, because === does NOT treat strings and numbers as being equal (i.e. === cares about type)
+
+c) true, for the same reason as a)
+
+d) false, because while x == "x" is true, x == "x " is false
+
+e) false, because the strings are different
+
+f) true, because the operator = outputs the input on the right and changes x to the value of the input on the right
+
+g) true, because declaring var x does not define it, and it remains undefined
+
+h) NaN, because < takes two numbers as its inputs and will not know how to evaluate two string inputs
+
+i) false, because typeof x is "number", and "number" + 1 is NaN, which is not the same as the string "number"
+
+j) false, for the same reason as i). "number" % 2 is NaN
+
+k) true when x is a number, because otherwise the result of x + 1 will not be a number
+
+l) true, because ++x converts x to a number and adds one, and x++ is the same as x = x + 1
+
```

Comment:
This will return false, due to mechanism in which post and pre addition is occurring to x. Consider this: x = 1. x++ will return the value of x (1) and then add 1 to that value. So what we are looking at is 1 == ++x(2), at this point ++x occurs, adding to x before returning x's value (which at this point is 3). So we're comparing 1 == 3, which is false.